src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java

```diff
@@ -69,6 +70,10 @@ public KNN80DocValuesProducer(DocValuesProducer delegate, SegmentReadState state
             if (!field.attributes().containsKey(KNN_FIELD)) {
                 continue;
             }
+            // Only segments that contains BinaryDocValues and doesn't have vector values
```

Comment:
elaborate the comment little bit more for our futurseleves.

Reply:
also lets add a UT for this

Reply:
Ack